python/pyarrow/includes/libarrow.pxd

```diff
@@ -19,6 +19,8 @@
 
 from pyarrow.includes.common cimport *
 
+from cpython.ref cimport PyObject
```

Comment:
I _think_ just declaring `object` works, no need to reach for this

Reply:
Yes you're right: https://github.com/cython/cython/issues/2589
I think you are referring to this idea.

Reply:
But I see in `arrow.pxi`, `PyObject*` is used to get the out variable. Why should we use `object` here?

Reply:
It seems `object` will manipulate the reference count for you, so `PyObject` is fine here then

Reply:
I think you can still pass `object`. `PyObject*` should ideally only be used as a special case, since it's less safe.